In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

Generally these Ebay auction sales are for sports autographs. 

In [2]:
df1 = pd.read_csv(r"raw/TestSet.csv")

In [5]:
# What is happening with TestSet.csv
df1.head()

EbayID  QuantitySold   Price  PricePercent  StartingBidPercent  \
0  400476160084             0    0.99        0.5892              0.5892   
1  350785007664             0  119.00        1.1427              1.1427   
2  380629932508             1    1.75        1.0416              0.4464   
3  300896813674             1   66.00        0.6338              0.0000   
4  200918931232             0    9.99        0.5197              0.5197   

           SellerName  SellerClosePercent  Category  PersonID  StartingBid  \
0          harryjean0            0.128269     73409      8215         0.99   
1          rrsports23            0.116667     27260        34       119.00   
2             mojo640            0.531599     73409      8215         0.75   
3      mintsignatures            0.971014     27260        34         0.01   
4  realdealsignatures            0.187117     27285     28504         9.99   

   ...  AuctionHitCountAvgRatio BestOffer  ReturnsAccepted  IsHOF  \
0  ...                        6  0.064103                0      1   
1  ...                       25  0.077922                0      1   
2  ...                        7  0.102564                0      0   
3  ...                       61  1.506494                0      1   
4  ...                        8  0.036364                0      1   

   ItemListedCount  AuctionCount  AuctionSaleCount  SellerAuctionCount  \
0                0            79                 7                2791   
1                1          1819               834                  60   
2                0            79                 7                 269   
3                1          1819               834                 138   
4                0           202                30                 326   

   SellerAuctionSaleCount  AuctionMedianPrice  
0                     358                0.99  
1                       7               77.00  
2                     143                0.99  
3                     134               77.00  
4                      61               11.00  

[5 rows x 28 columns]

In [9]:
df1.columns

Index(['EbayID', 'QuantitySold', 'Price', 'PricePercent', 'StartingBidPercent',
       'SellerName', 'SellerClosePercent', 'Category', 'PersonID',
       'StartingBid', 'AvgPrice', 'EndDay', 'HitCount', 'AuctionAvgHitCount',
       'ItemAuctionSellPercent', 'SellerSaleAvgPriceRatio', 'SellerAvg',
       'SellerItemAvg', 'AuctionHitCountAvgRatio', 'BestOffer',
       'ReturnsAccepted', 'IsHOF', 'ItemListedCount', 'AuctionCount',
       'AuctionSaleCount', 'SellerAuctionCount', 'SellerAuctionSaleCount',
       'AuctionMedianPrice'],
      dtype='object')

In [6]:
df1.shape

(37460, 28)

In [7]:
# What is happening with TrainingSubSet.csv
df2 = pd.read_csv(r"raw/TrainingSet.csv")
df2.head()

EbayID  QuantitySold  Price  PricePercent  StartingBidPercent  \
0  160983189073             1  27.25        1.4787              0.0537   
1  390549601720             1  43.00        1.4422              0.3350   
2  200901284676             1  11.00        0.5969              0.5426   
3  271162670857             0  10.00        0.5426              0.5426   
4  310611578768             0   9.99        1.9159              1.9159   

          SellerName  SellerClosePercent  Category  PersonID  StartingBid  \
0          petesandi            0.925926     73396      9174         0.99   
1         graphn4fun            0.990566     27278     11046         9.99   
2      hondo19461946            0.422131     73396      9174        10.00   
3        winemeyer45            0.245098     73396      9174        10.00   
4  chucksmemorabilia            0.295359     73396     17956         9.99   

   ...  AuctionHitCountAvgRatio BestOffer  ReturnsAccepted  IsHOF  \
0  ...                       29  1.540541                0      1   
1  ...                       55  0.404762                0      0   
2  ...                       21  0.324324                0      1   
3  ...                       26  0.270270                0      0   
4  ...                       53  0.200000                0      1   

   ItemListedCount  AuctionCount  AuctionSaleCount  SellerAuctionCount  \
0                1           583               291                  54   
1                0           120                43                 106   
2                1           583               291                 244   
3                1           583               291                 102   
4                0           137                44                 237   

   SellerAuctionSaleCount  AuctionMedianPrice  
0                      50               12.26  
1                     105               24.45  
2                     103               12.26  
3                      25               12.26  
4                      70                3.02  

[5 rows x 28 columns]

In [25]:
df2['AuctionSaleCount'].nunique()

402

In [26]:
df2['AuctionSaleCount']

0         291
1          43
2         291
3         291
4          44
         ... 
258583     17
258584     17
258585     17
258586      7
258587    128
Name: AuctionSaleCount, Length: 258588, dtype: int64

In [21]:
df2['BestOffer'].describe()

count    258588.000000
mean          0.268529
std           0.427103
min          -0.333333
25%           0.047619
50%           0.111111
75%           0.291667
max          18.750000
Name: BestOffer, dtype: float64

In [8]:
df2.shape

(258588, 28)

So the total number of records οn this dataset (both Test and Training Set) is 296048.<br>
The training subset is the 0.12 (12%) percentage of the whole dataset.<br>
The testing subset is the 0.88(88%) percentage of the whole dataset.<br>

# What does the meaning of each of those values ? <br>

EbayID --> Unique ID of each and every auction 

QuantitySold --> The number of items sold in the listing. Represented by a 0 or 1.

Price --> Final price the auction. If the listing does not result in a sale, the Price will be equal to the StartingBid. ,br

PricePercent --> 

StartingBidPercent --> The ratio of the StartingBid divided by sku’s AvgPrice.

SellerName --> The name of the person who put his aytographed item for sale.

SellerClosePercent --> The ratio of the number of auctions resulting in sale for a seller divided <br> by total number of auctions the seller listed

Category --> eBay autograph category by sport.

PersonID -->  ID of the person which is host in auction       ---------

StartingBid --> Minimum bid for the auction

AvgPrice --> The average sale price by sku (Stock Keeping Unit: μια μονάδα απογραφής. Υπάρχει μοναδικός αριθμός SKU για κάθε προιόν) 

EndDay --> The week days (Monday,Tuesday, Thursday ... Sunday) which happened the auction.

HitCount --> how many times the site of this speicific auction has been seen (πόσες φορες έχουν δει την δημοπρασια όσο πιο πολλές τόσο πιο δημοοφιλές) -----

AuctionAvgHitCount --> Avg hits for auctions of the same type or same product ----

ItemAuctionSellPercent --> The ratio of the number of auctions resulting in sale for a sku divided by total number of auctions the listed for the sku ----

SellerSaleAvgPriceRatio --> Average Price value for this specific product for specific seller  ----

SellerAvg --> Average Price value for this specific for specific seller  ------

SellerItemAvg --> Average Price value for specific seller   --------

AuctionHitCountAvgRatio --> Ratio of hits for a specific auction related to average hits for similar products

BestOffer --> Best bid on auction

ReturnsAccepted --> ratio of these items which takes best offer.

IsHOF --> IsHOF	Whether the player in their sport’s Hall of Fame. 

ItemListedCount -->

AuctionCount --> The number of auctions listed by sku 

AuctionSaleCount --> How many auctions have been completed with a sale  -----------

SellerAuctionCount -->   Πόσα auctions έχει δημιουργήσει ο συγκεκριμένος seller.-------------

SellerAuctionSaleCount --> 	The number of sales the seller has made

AuctionMedianPrice -->  Μεσαία τιμή (median) πώλησης για auctions του ίδιου SKU.------